packages/workspace/src/workspace/nacl_files/mutil_env/routers.ts

```diff
@@ -410,23 +454,34 @@ const addToSource = async (
     if (before === undefined) {
       return [createAddChange(wrappedElement, topLevelElement.elemID)]
     }
+
+    if (overrideTargetElements) {
+      // we want to override, not merge - so we need to wrap each gid individually
+      return _.flatten(gids.map(id => overrideIdInSource(
+        id,
+        before as ChangeDataType,
+        topLevelElement as ChangeDataType,
+      )))
+    }
+
     const mergeResult = mergeElements([
       before,
       wrappedElement,
     ])
-    // We should't have errors since before and wrapped elements are from common/env
-    // so they should be mergeable
     if (mergeResult.errors.length > 0) {
+      // If either the origin or the target source is the common folder, all elements should be
+      // mergeable and we shouldn't see merge errors
       throw new Error(
-        `Failed to move ${gids.map(id => id.getFullName())} Can not move unmergable element fragments.`
+        `Failed to add ${gids.map(id => id.getFullName())} - unmergable element fragments.`
       )
     }
     const after = mergeResult.merged[0] as ChangeDataType
     return detailedCompare(before, after, true)
   })))
-  return _.flatten(
-    await Promise.all(fullChanges.map(change => seperateChangeByFiles(change, originSource)))
-  )
+  return _.flatten(await Promise.all(fullChanges.map(change => seperateChangeByFiles(
+    change,
+    change.action === 'remove' ? targetSource : originSource
+  ))))
```

Comment:
nit (prefer builtin over lodash)
```suggestion
  return (await Promise.all(fullChanges.map(change => seperateChangeByFiles(
    change,
    change.action === 'remove' ? targetSource : originSource
  )))).flat()
```

Reply:
missed the comment about `seperateChangeByFiles` - will fix in https://github.com/salto-io/salto/pull/1302